image/helpers.go

```diff
@@ -404,3 +406,57 @@ func (tsto *ToolingStore) Find(at *asserts.AssertionType, headers map[string]str
 	}
 	return tsto.sto.Assertion(at, pk, tsto.user)
 }
+
+// var so that it can be mocked for tests
+var writeResolvedContent = writeResolvedContentImpl
+
+// XXX: move to gadget?
+func writeResolvedContentImpl(targetDir, prepareDir, gadgetUnpackDir, kernelUnpackDir string, model *asserts.Model) error {
+	fullPrepareDir, err := filepath.Abs(prepareDir)
+	if err != nil {
+		return err
+	}
+
+	info, err := gadget.ReadInfoAndValidate(gadgetUnpackDir, model, nil)
```

Comment:
there's an argument to be made that this function should receive the gadget.Info directly instead of the model.

the arguments would be:

targetDir, info, gadgetUnpackDir, kernelUnpackDir

Reply:
Indeed, I like that. I updated the code and had to tweak some tests. Note that the signature is slightly different (not targetDir) as I applied the earlier feedback about just using gadgetUnpackDir. Let me know if you feel it should be done differently.